packages/open-api-gateway/src/project/open-api-gateway-ts-project.ts

```diff
@@ -98,20 +98,22 @@ export class OpenApiGatewayTsProject extends TypeScriptProject {
 
   private readonly hasParent: boolean;
 
-  constructor(options: OpenApiGatewayTsProjectOptions) {
+  constructor(projectOptions: OpenApiGatewayTsProjectOptions) {
     super({
-      ...options,
+      ...projectOptions,
       sampleCode: false,
       tsconfig: {
-        ...options.tsconfig,
+        ...projectOptions.tsconfig,
         compilerOptions: {
           lib: ["dom", "es2019"],
-          ...options.tsconfig?.compilerOptions,
+          ...projectOptions.tsconfig?.compilerOptions,
         },
       },
     });
 
-    if (options.specFile) {
+    const options = this.preConstruct(projectOptions);
+
+    if (options.specFile && !path.isAbsolute(options.specFile)) {
```

Comment:
is the specFile is not absolute then we ignore the value and revert to spec.yaml?

Reply:
Is this documented?

Reply:
So if it's not absolute we'll still generate a sample hello world spec in that location relative to the project source dir, eg `specFile: 'my-folder/my-spec.yaml'` will result in the hello world spec being written to `src/my-folder/my-spec.yaml`.

Then if it's absolute we skip generating the sample and just use that as the input spec to parse :)

The docs say to use a path relative to the project source dir - using an absolute path directly would mean that users can't really share their code so I didn't want to document it as an available feature :)

Reply:
Makes sense.